update/Pos/Update/DB.hs

```diff
@@ -59,60 +64,70 @@ import           Pos.DB.Iterator            (DBIteratorClass (..), DBnIterator,
                                              runDBnMapIterator)
 import qualified Pos.DB.Limits              as DBLimits
 import           Pos.DB.Types               (NodeDBs (..))
-import           Pos.Genesis                (genesisBlockVersion, genesisBlockVersionData,
-                                             genesisSlotDuration, genesisSoftwareVersions)
 import           Pos.Slotting.Types         (EpochSlottingData (..), SlottingData (..))
-import           Pos.Types                  (ApplicationName, BlockVersion,
-                                             ChainDifficulty, NumSoftwareVersion, SlotId,
-                                             SoftwareVersion (..), Timestamp (..))
+import           Pos.Update.Constants       (ourAppName)
 import           Pos.Update.Core            (BlockVersionData (..), UpId,
                                              UpdateProposal (..))
+import           Pos.Update.Genesis         (UpdateGenesis (..))
 import           Pos.Update.Poll.Types      (BlockVersionState (..),
                                              ConfirmedProposalState,
                                              DecidedProposalState (dpsDifficulty),
                                              ProposalState (..),
                                              UndecidedProposalState (upsSlot),
                                              cpsSoftwareVersion, psProposal)
-import           Pos.Util                   (maybeThrow)
+import           Pos.Util.Context           (HasContext, getContext)
 import           Pos.Util.Iterator          (MonadIterator (..))
+import           Pos.Util.Util              (maybeThrow)
+
+-- | All of these instances are defined in cardano-sl, but not here, so we
```

Comment:
They should be moved into `cardano-sl-update`. I remember that we agreed about it. I don't remember whether it should be `Pos.Update.Binary*` or `Pos.Binary.Update*`.